policyengine_us/variables/gov/usda/snap/income/deductions/shelter/snap_excess_shelter_expense_deduction.py

```diff
@@ -25,7 +25,15 @@ def formula(spm_unit, period, parameters):
         housing_cost = add(
             spm_unit, period, ["snap_utility_allowance", "housing_cost"]
         )
-        uncapped_ded = max_(housing_cost - subtracted_income, 0)
+        # Exclude ineligible members' share per SNAP proration rules (result always >= 0 since prorate_fraction < spm_unit_size)
+        prorate_fraction = spm_unit("snap_prorate_fraction", period.this_year)
+        spm_unit_size = spm_unit("spm_unit_size", period)
+        housing_cost_after_proration = housing_cost * (
+            1 - prorate_fraction / spm_unit_size
```

Comment:
do we need to mask for spm unit size 0? There might be some households with 0 people in the data